[ACS][Common] OPS - Dual Persona

AikoBB opened the pull request:
# Description
This PR introduces changes to the ACS Common SDK, providing functionality to exchange Entra User tokens for ACS tokens. The key updates include:

- `CommunicationTokenCredential`: Extended with a new constructor that accepts - `EntraCommunicationTokenCredentialOptions`.

- `EntraTokenCredential`: Implementation where the actual token exchange is handled.

These changes enhance the SDK by enabling seamless integration of Entra User with Azure Communication Services.

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

Commit:
Merge remote-tracking branch 'upstream/main'

Commit:
Merge remote-tracking branch 'upstream/main'

Commit:
Merge remote-tracking branch 'upstream/main'

Commit:
Merge remote-tracking branch 'upstream/main'

Commit:
Merge remote-tracking branch 'upstream/main'

Commit:
Merge remote-tracking branch 'upstream/main'

Commit:
Merge remote-tracking branch 'upstream/main'

Commit:
Merge remote-tracking branch 'upstream/main'

Commit:
Merge remote-tracking branch 'upstream/main'

Commit:
Merge remote-tracking branch 'upstream/main'

Commit:
Add Entra token credentials

AikoBB committed:
fixed api comments & regenerated api

AikoBB committed:
added implementation

AikoBB committed:
Merge branch 'main' into feature/aigerimb/ops_dual_personas

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.Communication.Common](https://apiview.dev/Assemblies/Review/b818a161edd94363a3ca3c8893844a76?revisionId=7d2cf9f6354045a19f433bba1c5f12b7)

AikoBB committed:
fixed exchange implementation

AikoBB committed:
fixed docs

AikoBB committed:
fixed docs

AikoBB committed:
added token cache and covered with tests

AikoBB committed:
added read-me

nickliu-msft committed:
[Storage] [DataMovement] [API-Change] Changed ShareFilesStorageResourceProvider.FromFile/Directory to take a Uri over string (#46446)

* Initial commit

* Export API

SpencerOfwiti committed:
Update Nginx changelog to support API version 2024-06-01-preview (#46447)

* Update Nginx changelog to support API version 2024-06-01-preview

* Refactor

jsquire committed:
[Service Bus] October 2024 Release Prep (#46438)

The focus of these changes is to prepare the Service Bus package for the October
2024 release.

seanmcc-msft committed:
Storage STG 96 (#45604)

nielsheeren committed:
Fix missing HybridSearch in copy method of SearchOptions (#46425)

Fix missing HybridSearch in copy SearchOptions

AikoBB committed:
changed scopes to opt.param

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20241007.1 (#46457)

* Update Generator Version 3.0.0-beta.20241007.1

* Update SDK codes he_ma_3

* Update SDK codes de_he_2

* Update SDK codes ma_pu_4

* Update SDK codes sq_wo_6

* Update SDK codes co_de_1

* export api

* fix tests

---------

Co-authored-by: Arcturus Zhang <dapzhang@microsoft.com>
Co-authored-by: Minghao Chen <30464227+HarveyLink@users.noreply.github.com>

AikoBB committed:
fixed generated api

Sid200026 committed:
Onboard Azure.Developer.MicrosoftPlaywrightTesting sdk (#45044)

* feat(): onboard Azure.Developer.MicrosoftPlaywrightTesting sdk

* chore(): add support for github summary

* docs(): sample, CHANGELOG and README

* docs(): content of samples

* chore():: modify samples as per azure sdk guidelines

* refactor(): unit tests as per azure sdk guidelines

* chore(): use autorest to generate api clients

* fix(): json parsing of error messages

* chore(): add live tests

* fix(): run id handling across scalable and reporting

* Adressing comments

* comment

* chore(): resolve documentation comments

* docs(): import c# code snippets from .cs files

* refactor(): update CODEOWNERS with latest labels

* chore(): add git commit based display name

* feat(): scalable run error handling

* refactor(): convert API client class to internal

* chore(): apiview review comments addressed for base sdk

* refactor(): rename tokenCredential to credential

* refactor(): rename PlaywrightServiceSettings to PlaywrightServiceOptions

* refactor(): use serviceAuth only

* refactor(): remove defaultAuth references

* refactor(): convert serviceOs class to internal

* refactor(): convert public fields into properties

* chore(): add cancellation token in async methods

* refactor(): use default for null cancellation tokens

* refactor(): added users in CODEOWNERS

* fix(): live test resource json

* docs(): add missing sections in README files

* refactor(): rename base package to Azure.Developer.MicrosoftPlaywrightTesting.TestLogger

* docs(): move authenticate client section after prerequisites

---------

Co-authored-by: Siddharth Singha Roy <ssingharoy@microsoft.com>
Co-authored-by: Vansh Vardhan Singh <vanshvsingh@gmail.com>

DominikMe commented on sdk/communication/Azure.Communication.Common/src/EntraTokenCredential.cs:
```diff
@@ -0,0 +1,191 @@
+            return ParseAccessTokenFromResponse(message.Response);
+        }
+
+        private HttpMessage CreateRequestMessage()
```
We should reuse the Azure Core Http Pipeline (see the JS PR) so that we get all their retry logic etc

DominikMe commented on sdk/communication/Azure.Communication.Common/src/EntraTokenCredential.cs:
```diff
@@ -0,0 +1,191 @@
+        private HttpPipeline CreatePipelineFromOptions(EntraCommunicationTokenCredentialOptions options)
+        {
+            var authenticationPolicy = new BearerTokenAuthenticationPolicy(options.TokenCredential, options.Scopes);
+            return HttpPipelineBuilder.Build(ClientOptions.Default, authenticationPolicy);
```
Will this include the default retry pipeline option? Would be good to have a UT to validate.

AikoBB replied:
Yes, it will include [RetryPolicy](https://github.com/Azure/azure-sdk-for-net/blob/0598a8e8454c941051a7f9d323b09cfd3f18d893/sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs#L160C21-L160C32) in the HttpPipelineBuilder.Build call. Let me check if I can cover with test the retry part

AikoBB replied:
fixed

DominikMe commented on sdk/communication/Azure.Communication.Common/src/EntraTokenCredential.cs:
```diff
@@ -0,0 +1,191 @@
+                return _currentToken;
+
+            var shouldThisThreadExchange = false;
+            lock (_syncLock)
```
Have you looked into whether ThreadSafeRefreshableAccessTokenCache could be reused?

AikoBB replied:
Great suggestion! I'll see if I can reuse that logic.

AikoBB replied:
nope, it was not, fixed.

seanmcc-msft committed:
STG 96 beta changelogs (#46451)

* STG 96 beta changelogs

* PR comments

wps0 committed:
Azure.Monitor.Query: invalid date string format in MetricsClient (#45998)

azure-sdk committed:
Persist oidc env vars in deploy template (#46477)

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

azure-sdk committed:
Increment version for storage releases (#46479)

* Increment package version after release of Azure.Storage.Common

* Increment package version after release of Azure.Storage.Files.Shares

* Increment package version after release of Azure.Storage.Blobs

* Increment package version after release of Azure.Storage.Queues

* Increment package version after release of Azure.Storage.Blobs.Batch

* Increment package version after release of Azure.Storage.Files.DataLake

* Increment package version after release of Azure.Storage.Blobs.ChangeFeed

azure-sdk committed:
Increment package version after release of Azure.Messaging.ServiceBus (#46480)

jsquire committed:
[Event Hubs] Convert tests to use identity auth (#46330)

* [Event Hubs] Convert tests to use identity auth

The focus of these changes is to convert the Event Hubs
live tests to use identity-based authorization by default.
Connection strings will continue to have test coverage and
appear in "Hello World" samples and in legacy samples for
the migration guide.

* Test fixes

* Fixing more typos

* Test debugging

* tuning test timing and renabling cancellation  check

* tuning test timing

* Extending test timing slightly

* Update sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample02_EventHubsClientsLiveTests.cs

Co-authored-by: Madalyn Redding Heaps <66138537+m-redding@users.noreply.github.com>

* Update sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample02_EventHubsClientsLiveTests.cs

Co-authored-by: Madalyn Redding Heaps <66138537+m-redding@users.noreply.github.com>

* Update sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample02_EventHubsClientsLiveTests.cs

Co-authored-by: Madalyn Redding Heaps <66138537+m-redding@users.noreply.github.com>

* Update sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample02_EventHubsClientsLiveTests.cs

Co-authored-by: Madalyn Redding Heaps <66138537+m-redding@users.noreply.github.com>

* Update sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample02_EventHubsClientsLiveTests.cs

Co-authored-by: Madalyn Redding Heaps <66138537+m-redding@users.noreply.github.com>

* Update sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample02_EventHubsClientsLiveTests.cs

Co-authored-by: Madalyn Redding Heaps <66138537+m-redding@users.noreply.github.com>

* Update sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample02_EventHubsClientsLiveTests.cs

Co-authored-by: Madalyn Redding Heaps <66138537+m-redding@users.noreply.github.com>

* Update sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample02_EventHubsClientsLiveTests.cs

Co-authored-by: Madalyn Redding Heaps <66138537+m-redding@users.noreply.github.com>

* Regenerating snippets

---------

Co-authored-by: Madalyn Redding Heaps <66138537+m-redding@users.noreply.github.com>

jsquire committed:
[EngSys] Update central Service Bus version (#46483)

The focus of these changes is to update the central version for the
`Azure.Messaging.ServiceBus` package to use the latest release.

azure-sdk committed:
Use equality function check for persist oidc token step (#46485)

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 9106 (#46482)

Co-authored-by: Scott Beddall <scbedd@microsoft.com>

DominikMe commented on sdk/communication/Azure.Communication.Common/src/EntraTokenCredential.cs:
```diff
@@ -0,0 +1,191 @@
+        }
+
+        private bool IsTokenValid(AccessToken? token)
+            => token != null && DateTimeOffset.UtcNow < token?.ExpiresOn;
```
In my JS PR, if I remember correctly, I wanted to ensure that having a new Entra token will always get a new ACS token. That way the ACS token just follows whatever refresh logic the MSAL tokenCredential has internally.

AikoBB replied:
added the logic for cashing entra token

DominikMe commented on sdk/communication/Azure.Communication.Common/src/EntraTokenCredential.cs:
```diff
@@ -0,0 +1,191 @@
+        /// <returns> Contains the access token.</returns>
+        public AccessToken GetToken(CancellationToken cancellationToken)
+        {
+            return GetTokenAsync(cancellationToken).Result;
```
Let's use the EnsureCompleted Task extension instead of .Result

AikoBB replied:
fixed

AikoBB committed:
addressed comments

AikoBB committed:
addressed the latest comments

m-redding committed:
Update version of System.Text.Json (#46508)

dubiety committed:
Update Maps codeowners (#46494)

jalauzon-msft committed:
[Storage][DataMovement] Fix perf test cleanup + other minor fixes (#46486)

jaschrep-msft committed:
TransferManager tests (#46449)

* test queue failure

* More tests

* tests

* more tests (failing)

* test fixes. some new testing infra.

move one property access inside a trycatch

* reenable check

* remove enum to depend on test-wide def

m-redding committed:
Prep for release (#46512)

JonathanCrd committed:
Remove heaths as codeowner and add myself (#46488)

weshaggard committed:
Only treat critical vulnerabilities as errors (#46515)

nickliu-msft committed:
[Storage] [DataMovement] Moved all blob related tests out of DataMovement.Tests to DataMovement.Blobs.Test (#46475)

* Initial commit

* Moved more test files

* More tests moved over

* Compile Include over just adding the files to DM.Blobs.Tests

welovej committed:
Enable-WriteCore-feature-for-networkcloud (#46469)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20241009.1 (#46507)

* Update Generator Version 3.0.0-beta.20241009.1

* Update SDK codes he_ma_3

* Update SDK codes de_he_2

* Update SDK codes ad_co_0

* Update SDK codes pu_sp_5

* Update SDK codes co_de_1

* Update SDK codes ma_pu_4

* Update SDK codes sq_wo_6

* bump version of System.Text.Json

---------

Co-authored-by: Arcturus Zhang <dapzhang@microsoft.com>

frantran committed:
2024-05 Fleet .NET sdk generation (#46353)

* 2024-05 Fleet api .net sdk generation

* fix breaking change issue, add tests and changelog

* change changelog version

---------

Co-authored-by: frantran <ftran@microsoft.com>

Y-Sindo committed:
Support handling function return value from worker process (#46158)

Fix Azure/azure-functions-dotnet-worker#1496

The reason to add .NET 6 and .NET 8 as TFM:
The worker function return value is a Newtonsoft JObject if it's an object in the worker. Since .NET Core Frameworks, the default SignalR JSON protocol uses System.Text.Json library and can't serialize Newtonsoft JObject. Therefore, we need to wrap the JObject to a System.Text.Json serializable object with a customized JSON converter. The customized JSON converter first converts the Newtonsoft JObject to JSON with Newtonsoft library, then writes the raw JSON object with System.Text.Json writer. For .NET Standard 2.0, it's not possible to write a raw JSON with System.Text.Json writer, therefore, we need to add .NET 6 and .NET 8 as TFM to write the raw JSON.

archerzz committed:
fix(digitaltwins): bump System.Text.Json to 6.0.10 (#46523)

Co-authored-by: Mingzhe Huang (from Dev Box) <mingzhehuang@microsoft.com>

frantran committed:
add release date to Fleet .NET SDK latest version (#46524)

* add release date to latest version

* address version comment

* Update CHANGELOG.md

---------

Co-authored-by: frantran <ftran@microsoft.com>
Co-authored-by: Mingzhe Huang <archerzz@users.noreply.github.com>

DominikMe commented on sdk/communication/Azure.Communication.Common/src/EntraTokenCredential.cs:
```diff
@@ -0,0 +1,143 @@
+            }
+        }
+
+        private partial class AcsToken
```
Why is this partial?

AikoBB replied:
I do not even remember why I put partial here😁

DominikMe commented on sdk/communication/Azure.Communication.Common/tests/Identity/EntraTokenCredentialTest.cs:
```diff
@@ -0,0 +1,182 @@
+
+            // Act & Assert
+            var exception = Assert.ThrowsAsync<RequestFailedException>(async () => await entraTokenCredential.GetTokenAsync(CancellationToken.None));
+            Assert.AreEqual(true, exception?.Message.Contains(lastRetryErrorMessage));
```
Could we use equality instead of contains?

DominikMe commented on sdk/communication/Azure.Communication.Common/tests/Identity/EntraTokenCredentialTest.cs:
```diff
@@ -0,0 +1,182 @@
+                new MockResponse(500, "Second Error Message"),
+                new MockResponse(500, "Third Error Message"),
+                new MockResponse(500, lastRetryErrorMessage),
+                new MockResponse(500, "Fifth Error Message"),
```
Nice idea! Maybe let's change the fifth to "Fifth Error Message, shouldn't reach here" to make it very clear?

DominikMe commented on sdk/communication/Azure.Communication.Common/src/EntraTokenCredential.cs:
```diff
@@ -0,0 +1,143 @@
+                    ExchangeEntraToken,
+                    ExchangeEntraTokenAsync,
+                    false, null, null);
+        }
```
if we all agree to fetch immediately, we could call `GetTokenAsync` without awaiting it from within the constructor

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20241009.4 (#46526)

* Update Generator Version 3.0.0-beta.20241009.4

* Update SDK codes pu_sp_5

* Update SDK codes co_de_1

* regen

---------

Co-authored-by: Mingzhe Huang (from Dev Box) <mingzhehuang@microsoft.com>

azure-sdk committed:
Increment version for core releases (#46517)

* Increment package version after release of Azure.Core

* Increment package version after release of System.ClientModel

AikoBB committed:
added pre-warm logic

m-redding committed:
Update Packages.Data.props (#46534)

amnguye committed:
[Storage] Prepare for Storage Base Packages Preview Patch Release and WebJobs GA Patch Release (#46537)

* Prepare for Storage Patch Release

* Bumping version for Webjobs for GA patch release

JonathanCrd committed:
[Key Vault] Add CAE support & update System.ClientModel dependency version (#46013)

* Add flag and enable CAE to AuthorizeRequestInternal

* Enable CAE for AuthorizeRequestOnChallenge

* Add flag in SecretClientOption and SecretClient

* Revert "Add flag in SecretClientOption and SecretClient"

This reverts commit 02a805da8c3c4e48f2e863e1e455a96d9076dd7c.

* Enable CAE by default

* Removing unused parameter

* Remove saving the claims in the cache

* Update Changelog

* Update changelogs

* Simplify error checking logic

Addressing comment in PR

* Add test for base64 claims

* Override Process function to handle the first CAE Challenge after a scope challenge

* Add tests

* Separate credential and client transports and assert for a 401.

* Nest rety inside challenge if block

* Add test for claims in token

* Fix CI by removing extra test case parameter

* Nit changes to tests

* Simplify tests

* removing unnecessary mock responses

* Refactor tests to test CAE in all projects

* Make tests non parallelizable

* Add setup method to CAE tests

* Test for tokens obtained from cae challenges

* Fix test / CI

* Update dependency for System.ClientModel

* Apply suggestions

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20241010.1 (#46561)

* Update Generator Version 3.0.0-beta.20241010.1

* Update SDK codes de_he_2

* Update SDK codes ma_pu_4

* Update SDK codes sq_wo_6

azure-sdk committed:
Increment version for signalr releases (#46562)

melina5656 committed:
Enable WriteCore feature for storagecache (#46502)

* Enable WriteCore feature for storagecache

* Update assets.json

melina5656 committed:
Enable WriteCore feature for healthcareapis (#46491)

* Enable WriteCore feature for healthcareapis

* Update assets.json

* Update Azure.ResourceManager.HealthcareApis.netstandard2.0.cs

welovej committed:
Enable-WriteCore-feature-for-servicelinker (#46520)

melina5656 committed:
Enable WriteCore feature for trafficmanager (#46497)

melina5656 committed:
Enable WriteCore feature for storage (#46501)

welovej committed:
Enable-WriteCore-feature-for-servicefabricmanagedclusters (#46518)

welovej committed:
Enable-WriteCore-feature-for-servicebus (#46505)

welovej committed:
Enable-WriteCore-feature-for-securityinsights (#46504)

welovej committed:
Enable-WriteCore-feature-for-purview (#46498)

welovej committed:
Enable-WriteCore-feature-for-securitycenter (#46503)

welovej committed:
Enable-WriteCore-feature-for-privatedns (#46496)

welovej committed:
Enable write core feature for peering (#46493)

* Enable-WriteCore-feature-for-peering

* Update assets.json

welovej committed:
Enable-WriteCore-feature-for-orbital (#46492)

welovej committed:
Enable-WriteCore-feature-for-sphere (#46268)

* Enable-WriteCore-feature-for-sphere

* Update CHANGELOG.md

* Update CHANGELOG.md

welovej committed:
Enable-WriteCore-feature-for-recoveryservices-backup (#46243)

* Enable-WriteCore-feature-for-recoveryservices-backup

* Update assets.json

welovej committed:
Enable-WriteCore-feature-for-hdinsightcontainers (#46463)

welovej committed:
Enable-WriteCore-feature-for-mysql (#46466)

welovej committed:
Enable-WriteCore-feature-for-eventgrid (#46461)

welovej committed:
Enable-WriteCore-feature-for-labservices (#46464)

welovej committed:
Enable-WriteCore-feature-for-network (#46467)

welovej committed:
Enable-WriteCore-feature-for-resourcemover (#46499)

welovej committed:
Enable-WriteCore-feature-for-devtestlabs (#46455)

welovej committed:
Enable-WriteCore-feature-for-edgeorder (#46459)

welovej committed:
Enable-WriteCore-feature-for-springappdiscovery (#46521)

AikoBB committed:
add entra token cashing logic

Y-Sindo committed:
Update CODEOWNERS for SignalR(#46458)

nickliu-msft committed:
Initial commit (#46535)

jsquire committed:
[Event Hubs] Resilient management link creation (#46544)

The focus of these changes is to correctly handle the scenario where the
management AMQP link is in the process of closing (such as for an idle timeout)
as a new operation attempts to use it.  This error should be detected as a special
case and retried as it is with producer and consumer operations.

jamesvoongms committed:
Add generation of 2023-07-01-preview and new tests for dns resolver d… (#46105)

* Add generation of 2023-07-01-preview and new tests for dns resolver domain list and policy

* Add rule and link tests

* Fix test domain names and add to code owners and fix readme link to spec blob

* Fix CODEOWNERS for dns resolver

* Update .github/CODEOWNERS

Co-authored-by: Jesse Squire <jesse.squire@gmail.com>

* Update .github/CODEOWNERS

Co-authored-by: Jesse Squire <jesse.squire@gmail.com>

* Fix additional codeowner addition

* Add Servfail to cspell word list

* Run Export-Api

* Add assets file after pushing recording

* Record tests for dns resolver as well

* Change ActionType to DnsSecurityRuleActionTYpe

* Re-record DNS security rule tests after ActionType changes

* Change DNS resolver to DNS private resolver in codeowners

* Add new release version and notes

---------

Co-authored-by: James <james_voong@hotmail.com>
Co-authored-by: Jesse Squire <jesse.squire@gmail.com>

jsquire committed:
[CODEOWNERS] Adjust baseline errors (#46567)

The focus of these changes is to remove an account from the baseline errors, as the CODEOWNERS file has been updated to remove them.

Commit:
[AzureMonitorDistro] Fix livemetrics reporting incorrect values (#46429)

* removing buffer capacity

* remove counter

* remove DocumentBuffer class

* changelog

azure-sdk committed:
Increment version for storage releases (#46542)

Commit:
[AzureMonitorExporter] prep release 1.4.0 beta2 (#45906)

* disable tests in net6 and MacOS

* prep release

* update date

azure-sdk committed:
Increment version for storage releases (#46543)

tg-msft committed:
Azure.Provisioning: Update Redis to provide IsAccessKeyAuthenticationDisabled (#46568)

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 9102 (#46546)

* Add CompatibleConvertFrom-Yaml to common Package-Helpers
* Add CI artifact parsing to Save-Package-Properties, now each individual packageinfo.json file contains the relevant ci artifact lines for the package. This short circuits needing to re-parse this information in common checks.

---------

Co-authored-by: Scott Beddall <scbedd@microsoft.com>

azure-sdk committed:
Increment package version after release of Azure.Monitor.OpenTelemetry.Exporter (#46574)

Commit:
[AzureMonitorDistro] prep release 1.3.0 beta2 (#46572)

* update to new exporter beta

* update release date and changelog

welovej committed:
Enable-WriteCore-feature-for-fleet (#46462)

* Enable-WriteCore-feature-for-fleet

* regen

* Update assets.json

* Update Azure.ResourceManager.ContainerServiceFleet.netstandard2.0.cs

welovej committed:
Enable-WriteCore-feature-for-newrelicobservability (#46490)

* Enable-WriteCore-feature-for-newrelicobservability

* Update NewrelicManagementTestBase.cs

* Update assets.json

JoshLove-msft committed:
Prepare for release (#46576)

* Prepare for release

* version

rahuls-microsoft committed:
Update CODEOWNERS (#46575)

* Update CODEOWNERS

Adding @rahul-microsoft for fleet

* Update CODEOWNERS

Updating the alias

melina5656 committed:
Enable WriteCore feature for websubsub and release a new version (#46506)

* Enable WriteCore feature for webpubsub

* ignore WebAppWebPubSubConnectionCRUD()

* ignore WebAppKeyVaultConnectionCRUD()

* Update CHANGELOG.md

welovej committed:
Enable-WriteCore-feature-for-websites and release a new version (#46305)

* Enable-WriteCore-feature-for-websites

* Add temporary attribute to fix pipeline error

* release preparation

live1206 committed:
Bump version of MGC (#46565)

* Bump version of MGC

* add dependencies

* update lock file

* fix dependencies

* update test

* remove additional namespace

* specify peer dependencies

frantran committed:
update ContainerServiceFleet variables for more clear naming (#46582)

Co-authored-by: frantran <ftran@microsoft.com>

welovej committed:
Enable-WriteCore-feature-for-operationalinsights (#46225)

* Enable-WriteCore-feature-for-operationalinsights

* [Ignore("This test case will cause errors in the pipeline. After fixing the problem, restore this test")]

* Update SIAlertRuleCollectionTests.cs

* [Ignore("This test case will cause errors in the pipeline. After fixing the problem, restore this test")]

* [Ignore("This test case will cause errors in the pipeline. After fixing the problem, restore this test")]

* Update BookmarkResourceTests.cs

* [Ignore("This test case will cause errors in the pipeline. After fixing the problem, restore this test")]

melina5656 committed:
Enable WriteCore feature for containerservice (#46355)

* Enable WriteCore feature for containerservice

* ignore WebAppWebPubSubConnectionCRUD()

* Revert "ignore WebAppWebPubSubConnectionCRUD()"

This reverts commit 75d395dedba6e2fad9f4b26ef6231533e76cacd7.

* ignore FleetCRUDTest()

welovej committed:
Enable-WriteCore-feature-for-recoveryservices (#46242)

* Enable-WriteCore-feature-for-recoveryservices

* [Ignore("This test case will cause errors in the pipeline. After fixing the problem, restore this test")]

live1206 committed:
Update commit to fix tsp compile (#46587)

DominikMe commented on sdk/communication/Azure.Communication.Common/src/EntraTokenCredential.cs:
```diff
@@ -0,0 +1,145 @@
+        /// </summary>
+        /// <param name="options">The options for how the token will be fetched</param>
+        /// <param name="pipelineTransport">Only for testing.</param>
+        public EntraTokenCredential(EntraCommunicationTokenCredentialOptions options, HttpPipelineTransport pipelineTransport = null)
```
Can we make this constructor with the HttpPipelineTransport override internal, and add a public constructor that only takes in the options?

AikoBB replied:
EntraTokenCredential is an internal class, so that's why I went with having optional param instead of creating override. So can override it if you still suggest to override it 😁

DominikMe replied:
Ah, makes sense, I forgot that the whole class is internal.

DominikMe commented on sdk/communication/Azure.Communication.Common/tests/Identity/EntraTokenCredentialTest.cs:
```diff
@@ -0,0 +1,211 @@
+        }
+
+        [Test]
+        public async Task EntraTokenCredential_GetToken_MultipleCallsReturnsCachedToken()
```
Can we add a `EntraTokenCredential_GetToken_InternalEntraTokenChangeInvalidatesCachedToken` where we implement the `TokenCredential` in a way that it can be configured to return a different token? See my JS comment here: https://github.com/Azure/azure-sdk-for-js/pull/31291#discussion_r1796936348

AikoBB replied:
added

DominikMe commented on sdk/communication/Azure.Communication.Common/src/EntraTokenGuardPolicy.cs:
```diff
@@ -0,0 +1,50 @@
+
+namespace Azure.Communication
+{
+    internal class EntraTokenGuardPolicy : HttpPipelinePolicy
```
That's a cool idea!

DominikMe commented on sdk/communication/Azure.Communication.Common/src/EntraTokenGuardPolicy.cs:
```diff
@@ -0,0 +1,50 @@
+        {
+            var currentEntraToken = string.Empty;
+            message.Request.Headers.TryGetValue("Authorization", out currentEntraToken);
+            if (string.IsNullOrEmpty(_entraTokenCache) || currentEntraToken != _entraTokenCache)
```
If the previous response didn't succeed, then we need to make the request again if the token is unchanged, right?

AikoBB replied:
nice catch, will add one more check and cover with tests

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.WebPubSub (#46583)

azure-sdk committed:
Increment package version after release of Azure.Messaging.EventGrid (#46580)

seanmcc-msft committed:
Storage json hotfix (#46591)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.AppService (#46584)

kashish2508 committed:
fix(playwighttesting): Removed cancellationToken from SetupAsync method since NUnit does not allow additional params (#46566)

* Removed parameter from SetupAsync method.

* update

* undo last commit

* update

---------

Co-authored-by: guptakashish <guptakashish@microsoft.com>

nisha-bhatia committed:
Update test-resources.bicep (#46573)

KrzysztofCwalina committed:
Added the ability to provision AOAI as an optional component (#46570)

* fixed cminfra

* openai provisioning works

* open ai client added

* progress

* changed ai to key auth

* moved CM to WorkspaceClient abstraction

* refactored built-in methods and extension methods

* openai works

* updated api file

* disabled live tests

* updated version

* small tweaks

* updated api file

* PR feedback

* removed stj override

seanmcc-msft committed:
DM Lib beta release (#46596)

jsquire committed:
[EngSys] Bump package versions (#46595)

The focus of these changes is to bump packages that were flagged by component
governance this week related to the STJ 8.0.4 CVE.  Some older Event Hubs and
RC-based references were also bumped.

christothes committed:
Prepare for October GA release (#46600)

seanmcc-msft committed:
Fixed DM Lib Shares dependency (#46603)

Commit:
fix package dependency (#46609)

JonathanCrd committed:
[Key Vault] Prepare release for CAE feature (#46599)

* Revert "[KeyVault] Add support for service version 7.6-preview.1 (#43811)"

This reverts commit 699c7de7d021ff66327434ed53a357d95804547e.

* Run the prepare release script

kinelski committed:
[DocumentIntelligence] Default authentication to Token Credential (#46448)

azure-sdk committed:
Increment version for storage releases (#46610)

weshaggard committed:
Switch to TME test subscription (#46452)

melina5656 committed:
[.ContainerServiceFleet] Remove the ignore attribute and fix recording file (#46616)

welovej committed:
remove ignore attribute  and fix  the recording file  when Servicelinker project ref updated (#46588)

* Remove the ignore attribute from the test

* update1

---------

Co-authored-by: melina5656 <140602263+melina5656@users.noreply.github.com>

melina5656 committed:
Enable WriteCore feature for eventhub (#46169)

* Enable WriteCore feature for eventhub

* ignore CRUD() and AddRemoveTag(bool? useTagResource)

* ignore CreateOrUpdateExistGetGetAllDelete()

* Prepare for release

melina5656 committed:
Enable WriteCore feature for managedserviceidentity (#46563)

* Enable WriteCore feature for managedserviceidentity

* ignore UserAssignedEncryptionTests()

* Prepare for release

welovej committed:
Enable-WriteCore-feature-for-compute And Release (#46310)

* Enable-WriteCore-feature-for-compute

* Update assets.json

* Update assets.json

* Update assets.json

* Update CHANGELOG.md

* Update CHANGELOG.md

* [Ignore("This test case will cause errors in the pipeline. After fixing the problem, restore this test")]

* [Ignore("This test case will cause errors in the pipeline. After fixing the problem, restore this test")]

* [Ignore("This test case will cause errors in the pipeline. After fixing the problem, restore this test")]

* Revert "[Ignore("This test case will cause errors in the pipeline. After fixing the problem, restore this test")]"

This reverts commit d2645d7c000ac45d8885319a4b845dbf51c368e0.

* [Ignore("This test case will cause errors in the pipeline. After fixing the problem, restore this test")]

* release

welovej committed:
Enable-WriteCore-feature-for-containerregistry And Release (#46527)

* Enable-WriteCore-feature-for-containerregistry

* Update assets.json

* Update assets.json

* [Ignore("This test case will cause errors in the pipeline. After fixing the problem, restore this test")]

* [Ignore("This test case will cause errors in the pipeline. After fixing the problem, restore this test")]

* release

welovej committed:
Update CHANGELOG.md (#46625)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.ContainerRegistry (#46633)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.Compute (#46632)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.ManagedServiceIdentities (#46628)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.ContainerServiceFleet (#46623)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.EventHubs (#46627)

azure-sdk committed:
Increment package version after release of Azure.Identity (#46615)

scottaddie committed:
Delete DAC flow diagram (#46638)

azure-sdk committed:
Increment package version after release of Azure.Monitor.OpenTelemetry.AspNetCore (#46641)

azure-sdk committed:
Sync .github/workflows directory with azure-sdk-tools for PR 9134 (#46642)

* Use $GITHUB_EVENT_PATH variable instead of inline file writing

* Archive github event data

---------

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

azure-sdk committed:
Increment version for keyvault releases (#46647)

* Increment package version after release of Azure.Security.KeyVault.Certificates

* Increment package version after release of Azure.Security.KeyVault.Administration

* Increment package version after release of Azure.Security.KeyVault.Keys

* Increment package version after release of Azure.Security.KeyVault.Secrets

KrzysztofCwalina committed:
cleaned up CM workspace APIs (#46649)

* cleaned up workspace APIs

* PR feedback

melina5656 committed:
[HealthcareApis/ApiManagement] Remove the ignore attribute and fix recording file (#46629)

melina5656 committed:
[ServiceBus/EventHub] Remove the ignore attribute and fix recording file (#46630)

welovej committed:
update (#46634)

welovej committed:
Enable-WriteCore-feature-for-iothub And Release (#46178)

* Enable-WriteCore-feature-for-iothub

* [Ignore("This test case will cause errors in the pipeline. After fixing the problem, restore this test")]

* [Ignore("This test case will cause errors in the pipeline. After fixing the problem, restore this test")]

* [Ignore("This test case will cause errors in the pipeline. After fixing the problem, restore this test")]

* release

welovej committed:
Enable-WriteCore-feature-for-logic And Release (#46184)

* Enable-WriteCore-feature-for-logic

* [Ignore("This test case will cause errors in the pipeline. After fixing the problem, restore this test")]

* [Ignore("This test case will cause errors in the pipeline. After fixing the problem, restore this test")]

* release

welovej committed:
release (#46621)

welovej committed:
release (#46620)

AikoBB committed:
added acs token validity check

welovej committed:
[ContainerApps] Remove ingore attribute and fix the recording file after containerregistray release (#46635)

* update

* Update ContainerAppJobTests.cs

* Update ContainerAppJobTests.cs

* Update Packages.Data.props

DominikMe commented on sdk/communication/Azure.Communication.Common/tests/Pipeline/EntraTokenGuardPolicyTests.cs:
```diff
@@ -81,6 +85,60 @@ public async Task Process_WithDifferentToken_CallsProcessNext(bool async)
+            // Arrange
+            var policy = new EntraTokenGuardPolicy();
+            var pipeline = new Mock<HttpPipelinePolicy>();
+            var pipelines = new ReadOnlyMemory<HttpPipelinePolicy>(new HttpPipelinePolicy[] { _httpPipelinePolicyMock.Object });
```
Why do we need to use `ReadOnlyMemory` here?

AikoBB replied:
[HttpPipelinePolicy.ProcessNext](https://github.com/Azure/azure-sdk-for-net/blob/130d1c76e36b634d8087e0746a3e68c2f0287c43/sdk/core/Azure.Core/src/Pipeline/HttpPipelinePolicy.cs#L35) is expecting it as a param, and we can not anyhow mock it as it is a struct

DominikMe commented on sdk/communication/Azure.Communication.Common/tests/Pipeline/EntraTokenGuardPolicyTests.cs:
```diff
@@ -81,6 +85,60 @@ public async Task Process_WithDifferentToken_CallsProcessNext(bool async)
+        {
+            // Arrange
+            var policy = new EntraTokenGuardPolicy();
+            var pipeline = new Mock<HttpPipelinePolicy>();
```
Is `pipeline` being used anywhere? I don't see it.

DominikMe commented on sdk/communication/Azure.Communication.Common/tests/Pipeline/EntraTokenGuardPolicyTests.cs:
```diff
@@ -81,6 +85,60 @@ public async Task Process_WithDifferentToken_CallsProcessNext(bool async)
+            {
+                await policy.ProcessAsync(_httpMessage, pipelines);
+                await policy.ProcessAsync(_httpMessage, pipelines);
+                _httpPipelinePolicyMock.Verify(p => p.ProcessAsync(_httpMessage, It.IsAny<ReadOnlyMemory<HttpPipelinePolicy>>()), Times.Exactly(2));
```
IS this really verifying that the pipeline internally calls `ProcessNext`? At first read, it seems that it just verifies that `ProcessAsync` was called on the pipeline which happens in the two lines prior.

AikoBB replied:
So here we init `pipelines` 95th line with array of pipelines, which has only `_httpPipelinePolicyMock` pipeline. The `pipelines` is provided as a param to the `EntraTokenGuardPolicy. ProcessAsync` on 105th & 106th lines
<img width="1200" alt="image" src="https://github.com/user-attachments/assets/443dd507-4f81-4631-9bdd-067700d3340d">

inside the  `EntraTokenGuardPolicy` if the acs token invalid next pipeline is called and here we are forwarding `pipelines` param, and inside the [HttpPipelinePolicy.ProcessNext](https://github.com/Azure/azure-sdk-for-net/blob/130d1c76e36b634d8087e0746a3e68c2f0287c43/sdk/core/Azure.Core/src/Pipeline/HttpPipelinePolicy.cs#L35)  it will call the _httpPipelinePolicyMock
<img width="1023" alt="Pasted Graphic 1" src="https://github.com/user-attachments/assets/048f57de-767f-48d0-9a65-011a85c34a24">

@DominikMe would any further clarification be helpful?

DominikMe commented on sdk/communication/Azure.Communication.Common/src/EntraTokenGuardPolicy.cs:
```diff
@@ -46,5 +46,34 @@ private async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
+            message.Request.Headers.TryGetValue("Authorization", out currentEntraToken);
+            return (string.IsNullOrEmpty(_entraTokenCache) || currentEntraToken != _entraTokenCache, currentEntraToken);
+        }
+        private bool IsAcsTokenCacheOutdated()
```
nit: add empty line and it reads better to invert this to the positive phrasing of `IsAcsTokenCacheValid` (negate the bool condition accordingly).

DominikMe commented on sdk/communication/Azure.Communication.Common/src/EntraTokenGuardPolicy.cs:
```diff
@@ -46,5 +46,34 @@ private async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
             }
         }
+
+        private (bool CacheOutdated, string EntraToken) IsEntraTokenCacheOutdated(HttpMessage message)
```
It reads better to use positive language, i.e. `IsEntraTokenCacheValid` and `CacheValid` and only negate in the conditions as needed.

Eagle3386 committed:
Mark AzureAuthorityHosts.AzureGermany as Obsolete. (#46371)

scottaddie committed:
AzurePipelinesCredential doc updates (#46668)

annelo-msft committed:
Use BinaryData.Empty instead of bespoke solution in SCM and Azure.Core (#46669)

* Fix 41034

* update CHANGELOG

nickliu-msft committed:
[Storage] [DataMovement] Refactor Try-style methods (#46645)

* Initial commit

* Changed wording of doc comments

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.Logic (#46658)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.RecoveryServices (#46656)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.IotHub (#46655)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.OperationalInsights (#46657)

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 9147 (#46682)

* Changes required for sdk-repo based emitter pipelines

* Rename template

* Convert emitter archetype to pipeline template

* Move the archetype template out of the stages folder

---------

Co-authored-by: Patrick Hallisey <pahallis@microsoft.com>

azure-sdk committed:
Set resource group env variable early so remove always works (#46680)

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

azure-sdk committed:
Fix missing CloseAddressedIssues in gh event processor (#46678)

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

azure-sdk committed:
Handle missing artifacts without exception (#46676)

Co-authored-by: Scott Beddall <scbedd@microsoft.com>

whisper6284 commented on sdk/communication/Azure.Communication.Common/src/EntraTokenCredential.cs:
```diff
@@ -0,0 +1,145 @@
+        public ValueTask<AccessToken> GetTokenAsync(CancellationToken cancellationToken = default)
+            => _accessTokenCache.GetValueAsync(cancellationToken, () => true);
+
+        private AccessToken ExchangeEntraToken(CancellationToken cancellationToken)
```
I don't see anything calling this method?

AikoBB replied:
32th line, it's provided as a param to the ThreadSafeRefreshableAccessTokenCache init

whisper6284 commented on sdk/communication/Azure.Communication.Common/tests/Identity/EntraTokenCredentialTest.cs:
```diff
@@ -0,0 +1,231 @@
+        }
+
+        [Test]
+        public void EntraTokenCredential_InitWhtScopes_InitWithDefaultScope()
```
What's Wht?

AikoBB replied:
fixed

nicholasbergesen committed:
DevOpsInfrastructure - .Net SDK for GA stable version (#46652)

* Generate TypeSpec SDK files

* Update require in autorest.md

---------

Co-authored-by: Nicholas Bergesen <nbergesen@microsoft.com>

welovej committed:
update1 (#46659)

welovej committed:
update1 (#46660)

welovej committed:
update1 (#46662)

welovej committed:
Enable-WriteCore-feature-for-cosmosdb (#46137)

* Enable-WriteCore-feature-for-cosmosdb

* Update assets.json

melina5656 committed:
Update CHANGELOG.md (#46227)

kashish2508 committed:
chore(playwrighttesting): removed fallback authentication logic (#46667)

* Removed fallback authentication logic and modified ValidateMptPAT.

* updated FetchEntraIdAccessTokenAsync test

* updated test

* update

---------

Co-authored-by: guptakashish <guptakashish@microsoft.com>

yifanz7 committed:
[ElasticSan] Add support for ElasticSan auto scale policy (#46470)

AikoBB committed:
fixed comments

archerzz committed:
fix(samples): upgrade vulnerable version of `Azure.Storage.Blobs` (#46686)

Co-authored-by: Mingzhe Huang (from Dev Box) <mingzhehuang@microsoft.com>

HE-Xinyu committed:
K8s-runtime SDK (#45651)

azure-sdk committed:
Update AutoRest C# version to 3.0.0-beta.20241017.1 (#46690)

* Update Generator Version 3.0.0-beta.20241017.1

* Update SDK codes he_ma_3

* Update SDK codes de_he_2

* Update SDK codes co_de_1

* Update SDK codes ma_pu_4

* Update SDK codes sq_wo_6

* Update SDK codes ad_co_0

christothes committed:
Implement storage events (#46648)

sbomer committed:
Annotate Azure.Messaging.EventHubs for trim/AOT compat (#46606)

* Annotate Azure.Messaging.EventHubs for trim/AOT compat

* PR feedback

- Annotate Azure.Messaging.EventHubs.Processor, add ci check
- Add changelog entries

Also make some minor fixes to the AOT compat script.

* Fix RequiresUnreferencedCode message

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 9177 (#46670)

* Escape parentheses in includes to fix issues in publishing

* Only escape close parens

* Comments

---------

Co-authored-by: Daniel Jurek <djurek@microsoft.com>

weshaggard committed:
Fix storage tests-weekly matrix (#46651)

- Switch storage tests over to westus2 where we have VM capacity.

rhurey committed:
Update Readme.md and changelog for GA release. (#46684)

* Update Readme.md and changelog for GA release.

* Update changelog

christothes committed:
Fix issue where stand-alone ManagedIdentityCredential does not consider WorkloadIdentity (#46693)

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 9181 (#46695)

* save-package-properties.yml only activates PR diff mechanism if the service directory is "auto"

---------

Co-authored-by: Scott Beddall <scbedd@microsoft.com>

rahuls-microsoft committed:
Azure Compute Fleet GA version 2024-11-01 Minor update to Documentation only. (#46541)

yaotongms committed:
Release stable version 2024-07-10 of HybridCompute (#46581)

azure-sdk committed:
Update Generator Version 3.0.0-beta.20241017.2 (#46699)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.ContainerOrchestratorRuntime (#46706)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.HybridCompute (#46707)

azure-sdk committed:
Handle cleanup for leased and encrypted storage accounts. Reduce throttling (#46698)

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

glorialimicrosoft committed:
Added support for additional media message types: documents, videos, and audio (#46550)

* Added support for additional media message types: documents, videos, and audio.

* fix "Generated code is not up to date." error on the pipeline

* Try adding a general condition for all configurations for TreatWarningsAsErrors

* revert accidental check in of RecordedTestMode

* Adding customization code for classes using obsolete method and suppress the warning inline due to the defect here: https://github.com/Azure/autorest.csharp/issues/5114

* revert the unneeded csproj setting change

* update commit to latest in main

* Marking the packge for release

jaschrep-msft committed:
Merge transferjobinternal (#46540)

* setup of base method

* erase old impls

* remove child types

jsquire committed:
[CODEOWNERS] Removing invalid owners (#46710)

The focus of these changes is to remove accounts that have recently become invalid for CODEOWNER inclusion and are being flagged by the linter.

tg-msft committed:
Azure.Provisioning: Rename after review feedback (#46715)

Azure.Provisioning: Renames after review feedback

jalauzon-msft committed:
[Storage][DataMovement] Record some tests that could not be previously recorded (#46716)

tg-msft committed:
Azure.Provisioning: Unify resolvers and newslot Literal expressions (#46718)

azure-sdk committed:
Increment package version after release of Azure.Communication.Messages (#46713)

Sid200026 committed:
feat(playwrighttesting): separate api, test data processing and utility layers (#46681)

* feat(playwrighttesting): separate api, data processing and utility layers

* chore(): additional scalable run failure handling

* chore(): api retries and logging

* chore(): reporting api failure error handling

* fix(): no access token error message logged with error level

* chore(): remove null check condition in console writer

* chore(): add different logging levels

* refactor(): rename logger to microsoft-playwright-testing

* Update TestProcessor.cs

* Update ReporterUtils.cs

* Create ReporterUtilsTests.cs

* fix(): uri escape data string for workspace and run id

* fix(): ad hoc fixes

* fix(): handle null sas uri post refresh

* chore(): change reporting api version 2024-09-01-preview

* chore(): add tests for reporter

* chore(): add new api info

* fix(): remove parallelization from entra tests

---------

Co-authored-by: Siddharth Singha Roy <ssingharoy@microsoft.com>
Co-authored-by: Kashish Gupta <90824921+kashish2508@users.noreply.github.com>

Sid200026 committed:
docs(): update release date in CHANGELOG (#46730)

Co-authored-by: Siddharth Singha Roy <ssingharoy@microsoft.com>

Sid200026 committed:
fix(): remove dot from artifact safename (#46732)

Co-authored-by: Siddharth Singha Roy <ssingharoy@microsoft.com>

DominikMe commented on sdk/communication/Azure.Communication.Common/api/Azure.Communication.Common.netstandard2.0.cs:
```diff
@@ -53,6 +54,14 @@ public CommunicationUserIdentifier(string id) { }
+    public partial class EntraCommunicationTokenCredentialOptions
+    {
+        public EntraCommunicationTokenCredentialOptions(string resourceEndpoint, Azure.Core.TokenCredential entraTokenCredential) { }
+        public EntraCommunicationTokenCredentialOptions(string resourceEndpoint, Azure.Core.TokenCredential entraTokenCredential, string[] scopes) { }
```
Instead of adding a second constructor with scopes, we should add a setter to the Scopes property. Sorry for not catching that earlier.

DominikMe replied:
That's mainly to avoid having too many constructors when more properties get added over time and are optional.

AikoBB replied:
fixed, fetched the latest changes from the main branch and merged :)

DominikMe reviewed (approved):
Only one outstanding comment to make the readonly Scopes settable via property and not through the constructor, else lgtm!

hamshavathimunibyraiah committed:
Generating SDK with the latest TypeSpec commit (#46604)

* Updating with the latest TypeSpec commit ID

* Synced with main and re-generated SDK

* Fixing a build error, ran : ..\..\..\eng\scripts\Export-API.ps1 translation

* Adding Vikas as code owner

* Removing cpoulain

* Revert "Removing cpoulain"

This reverts commit 179d4a9d7412c74ae8f17db5f946f1e19e1dd4af.

* Updating the repo reference

* Removing cpoulain and this account lacks the required public membership to the Microsoft GitHub organization

tg-msft committed:
Azure.Provisioning: API cleanup (#46735)

- Removing features that won't be shipping as GA (like preview versions, `IClientCreator`, `StorageResources`, etc.) and hiding them behind a feature flag
- Fixes #46559 to correctly serialize ISO8601 duration `TimeSpan`s by adding a `Format` to `BicepValue`.
- Fixes #46558 to change string interpolation to use `InterpolatedStringHandler` and also add a `BicepStringBuilder`

christothes committed:
Let DAC chain continue for 400 responses with 'Identity not found' (#46711)

azure-sdk committed:
Move perf pipelines to TME subscription (#46739)

Co-authored-by: Wes Haggard <Wes.Haggard@microsoft.com>

jsquire committed:
[CODEOWNERS] Remove invalid account (#46741)

The focus of these changes is to remove an invalid account flagged by the linter.

tg-msft committed:
Azure.Provisioning: Avoid excess `null`s in BicepStringBuilder (#46742)

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 9229 (#46743)

* Remove default sub config sub-config-azure-cloud-test-resources

We no longer want to default the configuration and instead
default from the service connection which had the info we need.

* Fix quoting

---------

Co-authored-by: Wes Haggard <Wes.Haggard@microsoft.com>

emmeliaAra committed:
Update .NET SDK to remove property (#46272)

* Update .NET SDK to remove property

* update test cases

* update test

* update Assets.json test tag

* remove sessionRecords

* add session records back

* update

* update changelog.md file with new changes

* Update test cases

* update change log

* update tests

* update

* update

Han-msft committed:
AI Face SDK update  (#45402)

* Added support for the Large Face List and Large Person Group
* Added support for latest Detect Liveness Session API
* Change the default service API version to v1.2-preview.1.

Sid200026 committed:
fix(playwrighttesting): logger name and flaky tests (#46746)

* fix(): logger name and flaky tests

* fix(): run name fetching logic name from git

---------

Co-authored-by: Siddharth Singha Roy <ssingharoy@microsoft.com>

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.ComputeFleet (#46745)

SpencerOfwiti committed:
Update Nginx SDK version (#46758)

nickliu-msft committed:
Initial commit (#46759)

amruthashree18 committed:
Update for Azure Confidential Ledger Service Attention CODEOWNERS (#46760)

Update for Azure Confidential Ledger Service Attention CODEOWNERS

azure-sdk committed:
Fix spelling of Regenerate in yaml (#46761)

Co-authored-by: Patrick Hallisey <pahallis@microsoft.com>

azure-sdk committed:
Support branch name with space in create review step (#46763)

Co-authored-by: Praveen Kuttappan <prmarott@microsoft.com>

christothes committed:
Update MSAL to 4.66.1 (#46764)

audunn committed:
[NetAppFiles] update to 2024-07-01 (#46640)

* update to 2024-07-01

* update tests etc

* update assets

* update assets

* update assets

* update assets

* update assets

* update test

* update test

* update test

* update changelog

azure-sdk committed:
Increment package version after release of Azure.AI.Vision.Face (#46770)

melina5656 committed:
[CognitiveServices] Prepare to release a new version (#46701)

melina5656 committed:
[signalr] Prepare to release a new version (#46727)

Sid200026 committed:
docs(playwrighttesting): dotnet namespace fix and CHANGELOG update (#46771)

Co-authored-by: Siddharth Singha Roy <ssingharoy@microsoft.com>

live1206 committed:
Implement Response abstraction for Azure (#46530)

christothes committed:
Fix scenario where IMDS probe succeeds, but MI is unavailable (#46787)

hallipr committed:
Publish pipeline artifact with converted symbols (#46738)

* Publish pipeline artifact with converted symbols

* Use tempalteContect output

KrzysztofCwalina committed:
Fixed a bug in AZD integration code (#46791)

* improved code that rewrites appsettings.json

* removed unused namespaces

hallipr committed:
Fix symbol artifact output (#46797)

chunyu3 committed:
run Update-Mgmt-CI.ps1 in automation (#46748)

* run Update-Mgmt-CI.ps1 in automation

* Update eng/scripts/automation/GenerateAndBuildLib.ps1

Co-authored-by: Wes Haggard <weshaggard@users.noreply.github.com>

---------

Co-authored-by: Wes Haggard <weshaggard@users.noreply.github.com>

wangbwn committed:
DataFactory DotNet SDK Release (#46744)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.DataFactory (#46803)

anaarias-msft committed:
AVS 2023-09-01 SDK generation (#46324)

ArthurMa1978 committed:
Prepare release for Azure.ResourceManager.ContainerInstance 1.3.0-beta.1 (#46804)

Sid200026 committed:
chore(): 1.0.0-beta.1 post release changes (#46809)

Co-authored-by: Siddharth Singha Roy <ssingharoy@microsoft.com>

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.ElasticSan (#46812)

puagarwa committed:
feat(playwrighttesting): Add support to publish attachments (#46780)

* Add support to upload attachments

* fix tests

* minor comments fixes

azure-sdk committed:
Sync eng/common directory with azure-sdk-tools for PR 9236 (#46818)

* Refactor PackageProps to make the Yaml loading and processing methods common

* Updates for feedback

---------

Co-authored-by: James Suplizio <jasupliz@microsoft.com>

christothes committed:
Prep Identity for 1.13.1 patch release (#46817)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.ContainerInstance (#46810)

azure-sdk committed:
Increment package version after release of Azure.ResourceManager.Avs (#46811)

trangevi committed:
[Azure.AI.Inference] Beta 2 (#45757)

Features Added

Added new EmbeddingsClient, to provide support for generating text embeddings using supported models.
Add support for passing a string file path on disk in order to provide an image for chat completions.

Breaking Changes

ChatCompletionsClientOptions has been renamed to AzureAIInferenceClientOptions.
ChatCompletions response object has been flattened. ChatCompletions.Choices has been removed, and the underlying properties have been bubbled up to be on the ChatCompletions object instead.
ChatCompletionsFunctionToolCall has been replaced with ChatCompletionsToolCall.
ChatCompletionsFunctionToolDefinition has been replaced with ChatCompletionsToolDefinition.
ChatCompletionsToolSelectionPreset has been replaced with ChatCompletionsToolChoicePreset.
ChatCompletionsNamedFunctionToolSelection has been replaced with ChatCompletionsNamedToolChoice.
ChatCompletionsFunctionToolSelection has been replaced with ChatCompletionsNamedToolChoiceFunction.

Bugs Fixed

Fixed support for chat completions streaming while using tools.

Other Changes

Removed the need to manually provide an api-key header when talking to Azure OpenAI.

----------------------------------------------------------

Signed-off-by: Travis Angevine <trangevi@microsoft.com>
Co-authored-by: Liudmila Molkova <limolkova@microsoft.com>

jalauzon-msft committed:
[Storage][DataMovement] Add perf tests for DMLib Track1 (#46768)

calvinkwtang committed:
Adds SensitivityMask to DialogSensitivityUpdate (#46796)

azure-sdk committed:
Increment package version after release of Azure.AI.Vision.ImageAnalysis (#46820)

azure-sdk committed:
Increment package version after release of Azure.AI.Inference (#46821)

azure-sdk committed:
Allow default matrix selection override (#46825)

Co-authored-by: Ben Broderick Phillips <bebroder@microsoft.com>

welovej committed:
Enable-WriteCore-for-Azure.Developer.DevCenter (#46779)

AikoBB committed:
added setter for scopes

AikoBB committed:
small test fix

AikoBB committed:
Merge remote-tracking branch 'upstream/main' into feature/aigerimb/ops_dual_personas
